### ISSUE TITLE
Clean unused code in Development

### ISSUE THREAD

**Issue by jan-petr** [ExploreASL/ExploreASL#451]
**This initial comment should contain all information - please reflect decisions reached in the discussion below in this list**
I have noticed that while Functions folder looks pretty good, the Development folder is a huge mess. So I propose that we at least try to clean old functions that are no longer used and will never be. We can either completely delete them, or move them to Custom scripts. This will make the code cleaner and easier to maintain.

We don't have to do everything, but at least the obvious files in the first round. I have checked all those below and here are my propositions:

REMOVE:
- `Development/BIDS/ZeroingOut.m` (:+1:)
- `Development/media/CreateMeanControl.m` (:+1:)
- `Development/media/CreateGif_Averaging.m` (:+1:)
- `Development/media/CreateGif_Averaging_4D_template.m` (:+1:)
- `CalculateDistanceFromCenter.m` (:+1:)

MOVE TO THE ROOT DEVELOPMENT FOLDER:
- `Development/ExploreASL_make_standalone/ConvertNiiGz2Mat.m` (:+1:)

LEGACY. These functions appear to be no longer used, but they are nicely written, not a mess. So lets make a folder CustomScripts/ExploreASLLegacy and move those there for now:
- `Development/xASL_bids_DataPar2JSON.m`  ( :+1: in #62)
- `Development/xASL_bids_InsertJSONFields.m` (:+1:)
- `Development/SaraElgayar_BIDS` (:+1:)

PROMOTE. Those functions do not longer need to be in the Development folder and can be moved to functions. But before that, the headers should be cleaned and the whole code roughly checked.
- `xASL_test_BIDSFlavorsFull` ( :+1: in #394)
- `xASL_im_HausdorffDist` (:+1:)
- `xASL_bids_TestBidsConversion` ( :+1: in #290)

### Release note
Clean unused code and cleaned the development directory.

**Comment by MichaelStritt** [comment]
For `xASL_bids_TestBidsConversion` we should probably find a nice way to do this with regard to #290.

**Comment by jan-petr** [comment]
> For `xASL_bids_TestBidsConversion` we should probably find a nice way to do this with regard to #290.

I agree to do that in #290 . If you do so, just make a note in the original commit at the top so that we have the list up-to-date :)

Can we get rid of the 'make_standalone' functions?

**Comment by MichaelStritt** [comment]
> Can we get rid of the 'make_standalone' functions?

I think so, Henk wrote them a while ago when he tried to make it work on mac. Now it doesn't use them as far as I know.

**Comment by HenkMutsaerts** [comment]
* `Development/ExploreASL_make_standalone/ConvertNiiGz2Mat.m` we use if we have new .nii.gz atlases, that need to convert in a .mat sidecar for ExploreASL to use without needing to extract the .nii.gz.
* `Development/Import/xASL_ImportFull.m` I believe this is a mini pipeline I created for the old import. Note that it links to a code that moves DICOMs to a folder with the name of the SeriesDescription, that is a necessary preparation for the old import script (and probably the new one as well :)
* Sara Elgayar put quite some time in her BIDS parsing, and there are some good scripts in there

So I would move the three above to a /CustomScripts/ExploreASLLegacyCode folder.
The rest can be deleted.

`make_standalone`: actually @MichaelStritt created this folder :)
* So we need the `ConvertNiiGz2Mat.m` script, `xASL_spm_jobman.m` we need if we compile, as ExploreASL compiling didn't work sometimes without compiling SPM as well? Or I used this only for testing. The `*.bat` script was for testing on Windows, can be deleted. I have never compiled for Mac ;)
